src/app/pages/send-tokens/send-tokens.tsx

```diff
@@ -55,6 +57,10 @@ function SendTokensFormBase() {
     }
   }, [navigate, showNetworks]);
 
+  useEffect(() => {
+    if (!currentAccountNonce) setCurrentAccountNonce(nonce);
+  }, [currentAccountNonce, nonce, setCurrentAccountNonce]);
```

Comment:
This code isn't super clear. We pull in two nonce values, if if one doesn't exist, we use the other. 

Could this be wrapped as a separate hook that returns a single nonce? Why wouldn't `useNextNonce` itself call `setCurrentAccountNonce`?

Reply:
`currentAccountNonce` is setting an atom that is used in other atoms. The nonce here is the queried nonce, so it is set so it can be used in the atoms. If is the same as the pattern we use where we are setting atoms in the query hooks.

Reply:
I'll look at the few places in atoms we use the nonce, maybe I can totally remove needing any atom for the nonce. Or, I can look at moving setting it into the query hook like we do with other atoms.

Reply:
Why no deduplicate the code?

```ts
export function useSetCurrentAccountNonce() {
  const nonce = useNextNonce();

  useEffect(() => {
    if (!currentAccountNonce) setCurrentAccountNonce(nonce);
  }, [currentAccountNonce, nonce, setCurrentAccountNonce]);

  return nonce;
}
```

It's not exactly the same, either. In other cases, we use `onSuccess` right, so it's only ever set once in the hook, rather than in each instance of it's use.